tasks/eva_2850/fix_discordant_variants.py

```diff
@@ -0,0 +1,337 @@
+
+                    if rs not in all_rs_variants:
+                        logger.error(f"No RS variant could be found for RS {rs}")
+                        # If no RS found in DB, check if the original RS has been merged to some other RS,
+                        # if yes, add the new RS to the list for processing
+                        rs_events = all_events[rs]
+                        for event in rs_events:
+                            if event['accession'] == rs and event['eventType'] == 'MERGED':
+                                merge_into_RS = event['mergeInto']
+                                logger.error(
+                                    f"RS {rs} has been merged into RS {merge_into_RS}. Adding {merge_into_RS} to the list")
+                                rs_list_to_process.append(rs)
+                                break
+                        continue
+
+                    rs_records = all_rs_variants[rs]
+
+                    rs_without_map_weight = get_rs_without_map_weight(rs_records)
+                    if not rs_without_map_weight:
+                        logger.error(f"All variants for {rs} are map-weighted : \n{rs_records}")
+                        continue
+                    elif len(rs_without_map_weight) > 1:
+                        logger.error(f"More than one variant without map-weight found for RS {rs} :"
+                                     f"\n {rs_without_map_weight}")
+                        continue
+
+                    ss_records = all_ss_variants[rs]
+                    if not ss_records:
+                        logger.error(f"No original SS record found for RS {rs}, /n {ss_records}")
+                        continue
+
+                    rs_variant = rs_records[0]
+
+                    if check_all_ss_has_same_info(ss_records):
+                        if rs_variant['start'] == ss_records[0]['start']:
+                            logger.error(f"RS's and original SS's Start matches. Nothing to do")
+                            continue
+
+                        process_all_ss_has_same_info(rs_variant, ss_records)
+
+                    else:
+                        logger.error(
+                            f"For RS {rs}, Not all original SS has same info. Case for Split: \nSS Records {ss_records}")
+
+                rs_list = rs_list_to_process.copy()
+                rs_list_to_process.clear()
+
+
+
+
+def process_all_ss_has_same_info(rs_variant, ss_records):
+    rs_with_new_start = copy.copy(rs_variant)
+    rs_with_new_start['start'] = ss_records[0]['start']
+    rs_with_new_start['_id'] = get_clustered_SHA1(rs_with_new_start)
+
+    variant_in_db = check_for_hash_collision(rs_with_new_start['_id'])
+
+    if variant_in_db:
+        logger.warn(f"Hash collision will occur for RS {rs_variant['accession']} with RS {variant_in_db['accession']}")
+        resolve_collision_and_insert_rs(rs_variant, rs_with_new_start, variant_in_db)
```

Comment:
Since you have to do inserts anyway for this case and the else case below, suggest making it a separate function.

Reply:
updated